File: backend/src/main/java/com/corpomotriz/repuestos/model/Usuario.java
package com.corpomotriz.repuestos.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "usuario", schema = "repuestos_db")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
@ToString
public class Usuario {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nombre;
    private String email;
    private String contrasena;
    private String rol;

    @Column(name = "fecha_creacion")
    private LocalDateTime fechaCreacion;
}

File: frontend/src/app/model/usuario.ts
export interface Usuario {
  id: number;
  nombre: string;
  email: string;
  rol: string;
  fechaCreacion: string | Date;
}

export interface UsuarioCrearDTO {
  nombre: string;
  email: string;
  contrasena: string;
  rol: string;
}
